chainerio/filesystems/hdfs.py

```diff
@@ -62,6 +62,11 @@ def _wrap_fileobject(self, file_obj: Type['IOBase'],
 
         if 'b' not in mode:
             file_obj = io.TextIOWrapper(file_obj, encoding, errors, newline)
+        elif 'r' in mode:
+            # wrap the file_obj with io.BufferedReader to add `peek` support,
+            # which signiciantly improve pickle performance.
+
+            file_obj = io.BufferedReader(file_obj)
```

Comment:
Don't you work on 'r' mode for `io.BufferedWriter()` ? Could be a low-hanging fruit and worth give it a try.

Reply:
Sure